File: lib/layer2_core/dark/e_primary_fixed_dim_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EPrimaryFixedDimDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd21;
      case 'standard': return cstd21;
      case 'blue': return cbd21;
      case 'green': return cgd21;
      case 'red': return crd21;
      case 'nothing': return cnod21;
      default: return cstd21;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EPrimaryFixedDimDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EPrimaryFixedDimDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd21
            AppTheme.STANDARD -> R.color.cstd21
            AppTheme.BLUE -> R.color.cbd21
            AppTheme.GREEN -> R.color.cgd21
            AppTheme.RED -> R.color.crd21
            AppTheme.NOTHING -> R.color.cnod21
        }
    }
}
